config/shared_config.go

```diff
@@ -232,58 +229,75 @@ func loadSharedConfig(ctx context.Context, configs configs) (Config, error) {
 		}
 	}
 
-	return NewSharedConfig(ctx, profile, sharedConfigsLoadInfo{
-		CredentialsFiles:   credentialsFiles,
-		ConfigurationFiles: configurationFiles,
-		Logger:             logger,
-	})
+	return LoadSharedConfigProfile(ctx, profile,
+		func(o *LoadSharedConfigOptions) {
+			o.Logger = logger
+			o.ConfigFiles = configFiles
+			o.CredentialsFiles = credentialsFiles
+		},
+	)
 }
 
-// sharedConfigsLoadInfo struct contains Values that can be used to load the config.
-type sharedConfigsLoadInfo struct {
+// LoadSharedConfigOptions struct contains optional values that can be used to load the config.
+type LoadSharedConfigOptions struct {
 
 	// CredentialsFiles are the shared credentials files
 	CredentialsFiles []string
 
-	// ConfigurationFiles are the shared config files
-	ConfigurationFiles []string
+	// ConfigFiles are the shared config files
+	ConfigFiles []string
 
 	// Logger is the logger used to log shared config behavior
 	Logger logging.Logger
 }
 
-// NewSharedConfig retrieves the configuration from the list of files
+// LoadSharedConfigProfile retrieves the configuration from the list of files
 // using the profile provided. The order the files are listed will determine
 // precedence. Values in subsequent files will overwrite values defined in
 // earlier files.
 //
 // For example, given two files A and B. Both define credentials. If the order
 // of the files are A then B, B's credential values will be used instead of A's.
-func NewSharedConfig(ctx context.Context, profile string, options sharedConfigsLoadInfo) (SharedConfig, error) {
-	if len(options.ConfigurationFiles) == 0 && len(options.CredentialsFiles) == 0 {
-		return SharedConfig{}, fmt.Errorf("no shared config or shared credentials options provided")
+//
+// If no config files, credentials files are provided, the LoadSharedConfigProfile
+// will default to location `.aws/config` for config files and `.aws/credentials`
+// for credentials files respectively as per
+// https://docs.aws.amazon.com/credref/latest/refdocs/file-location.html#file-location
+//
+func LoadSharedConfigProfile(ctx context.Context, profile string, optFns ...func(*LoadSharedConfigOptions)) (SharedConfig, error) {
+	var option LoadSharedConfigOptions
+	for _, fn := range optFns {
+		fn(&option)
+	}
+
+	if len(option.ConfigFiles) == 0 {
+		option.ConfigFiles = DefaultSharedConfigFiles
+	}
+
+	if len(option.CredentialsFiles) == 0 {
+		option.CredentialsFiles = DefaultSharedCredentialsFiles
 	}
```

Comment:
suggest doing nil check here instead of len so caller could specify an empty list for the `ConfigFiles`/`CredentialsFiles`